scripts/packages/VSCodeServer/src/VSCodeServer.jl

```diff
@@ -193,126 +197,181 @@ function serve(args...; is_dev = false, crashreporting_pipename::Union{AbstractS
     conn_endpoint[] = JSONRPC.JSONRPCEndpoint(conn, conn)
     run(conn_endpoint[])
 
-    if is_dev
-        @async while true
-            try
-                Base.invokelatest(handle_message)
-            catch err
-                Base.display_error(err, catch_backtrace())
-            end
-        end
-    else
-        @async try
-            while true
-                Base.invokelatest(handle_message; crashreporting_pipename = crashreporting_pipename)
+    @async try
+        msg_dispatcher = JSONRPC.MsgDispatcher()
+
+        msg_dispatcher[repl_runcode_request_type] = repl_runcode_request
+        msg_dispatcher[repl_getvariables_request_type] = repl_getvariables_request
+        msg_dispatcher[repl_getlazy_request_type] = repl_getlazy_request
+        msg_dispatcher[repl_showingrid_notification_type] = repl_showingrid_notification
+        msg_dispatcher[repl_loadedModules_request_type] = repl_loadedModules_request
+        msg_dispatcher[repl_isModuleLoaded_request_type] = repl_isModuleLoaded_request
+        msg_dispatcher[repl_startdebugger_notification_type] = (conn, params)->repl_startdebugger_request(conn, params, crashreporting_pipename)
+
+        while true
+            msg = JSONRPC.get_next_message(conn_endpoint[])
+
+            if is_dev
+                try
+                    JSONRPC.dispatch_msg(conn_endpoint[], msg_dispatcher, msg)
+                catch err
+                    Base.display_error(err, catch_backtrace())
+                end
+            else
+                JSONRPC.dispatch_msg(conn_endpoint[], msg_dispatcher, msg)
             end
-        catch err
-            Base.display_error(err, catch_backtrace())
         end
+    catch err
+        # Add crash reporting
+        Base.display_error(err, catch_backtrace())
     end
 end
 
-function handle_message(; crashreporting_pipename::Union{AbstractString,Nothing}=nothing)
-    msg = JSONRPC.get_next_message(conn_endpoint[])
-    if msg["method"] == "repl/runcode"
-        params = msg["params"]
+JSONRPC.@dict_readable struct ReplRunCodeRequestParams <: JSONRPC.Outbound
```

Comment:
Do you think it would make sense to automatically add a kwarg constructor here? IMHO it's very convenient to not have to remember argument order.

Reply:
Heh, nvm: https://github.com/julia-vscode/JSONRPC.jl/issues/20.